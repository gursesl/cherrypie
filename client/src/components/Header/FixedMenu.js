import React from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import { Container, Menu } from 'semantic-ui-react'
import renderButtons from './renderButtons'

const FixedMenu = data => (
  <Menu fixed="top" size="large">
    <Container>
      <Menu.Item as={NavLink} to="/" exact>
        Home
      </Menu.Item>
      <Menu.Item as={NavLink} to="/users">
        Users
      </Menu.Item>
      <Menu.Item as={NavLink} to="/gqlusers">
        GQL Users
      </Menu.Item>
      <Menu.Item as={NavLink} to="/counter">
        Counter
      </Menu.Item>
      <Menu.Item as={NavLink} to="/weather">
        Weather
      </Menu.Item>
      <Menu.Menu position="right">{renderButtons(data)}</Menu.Menu>
    </Container>
  </Menu>
)

export default withRouter(FixedMenu)
