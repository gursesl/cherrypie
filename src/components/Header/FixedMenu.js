import React from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import {
  Button,
  Container,
  Menu,
} from 'semantic-ui-react'
import LoginModal from '../LoginModal'
import LogoutModal from '../LogoutModal'

const FixedMenu = () => (
  <Menu fixed="top" size="large">
    <Container>
      <Menu.Item as={NavLink} to="/" exact>Home</Menu.Item>
      <Menu.Item as={NavLink} to="/users">Users</Menu.Item>
      <Menu.Item as={NavLink} to="/counter">Counter</Menu.Item>
      <Menu.Item as={NavLink} to="/weather">Weather</Menu.Item>
      <Menu.Menu position="right">
        <Menu.Item position="right">
          <LoginModal />
          <LogoutModal />
        </Menu.Item>
      </Menu.Menu>
    </Container>
  </Menu>
)

export default withRouter(FixedMenu)
