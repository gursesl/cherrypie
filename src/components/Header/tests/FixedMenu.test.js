import React from 'react'
import { shallow } from 'enzyme'
import '../../../setupTests'
import FixedMenu from '../FixedMenu'

describe('FixedMenu', () => {
  const container = shallow(<FixedMenu.WrappedComponent />)
  it('should render without blowing up', () => {
    expect(container).toBeDefined()
  })

  it('renders itself into the DOM with the correct shallowComponent styles', () => {
    expect(container).toMatchSnapshot();
  });
})
