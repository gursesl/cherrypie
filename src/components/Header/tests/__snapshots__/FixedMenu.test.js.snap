// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedMenu renders itself into the DOM with the correct shallowComponent styles 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FixedMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <MenuItem
          as={[Function]}
          exact={true}
          to="/"
        >
          Home
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/users"
        >
          Users
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/gqlusers"
        >
          GQL Users
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/counter"
        >
          Counter
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/weather"
        >
          Weather
        </MenuItem>
        <MenuMenu
          position="right"
        >
          <MenuItem
            position="right"
          >
            <LoginModal />
            <LogoutModal />
          </MenuItem>
        </MenuMenu>
      </Container>,
      "fixed": "top",
      "size": "large",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            as={[Function]}
            exact={true}
            to="/"
          >
            Home
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/users"
          >
            Users
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/gqlusers"
          >
            GQL Users
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/counter"
          >
            Counter
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/weather"
          >
            Weather
          </MenuItem>,
          <MenuMenu
            position="right"
          >
            <MenuItem
              position="right"
            >
              <LoginModal />
              <LogoutModal />
            </MenuItem>
          </MenuMenu>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Users",
            "to": "/users",
          },
          "ref": null,
          "rendered": "Users",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "GQL Users",
            "to": "/gqlusers",
          },
          "ref": null,
          "rendered": "GQL Users",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Counter",
            "to": "/counter",
          },
          "ref": null,
          "rendered": "Counter",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Weather",
            "to": "/weather",
          },
          "ref": null,
          "rendered": "Weather",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <MenuItem
              position="right"
            >
              <LoginModal />
              <LogoutModal />
            </MenuItem>,
            "position": "right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <LoginModal />,
                <LogoutModal />,
              ],
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <MenuItem
            as={[Function]}
            exact={true}
            to="/"
          >
            Home
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/users"
          >
            Users
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/gqlusers"
          >
            GQL Users
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/counter"
          >
            Counter
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/weather"
          >
            Weather
          </MenuItem>
          <MenuMenu
            position="right"
          >
            <MenuItem
              position="right"
            >
              <LoginModal />
              <LogoutModal />
            </MenuItem>
          </MenuMenu>
        </Container>,
        "fixed": "top",
        "size": "large",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              as={[Function]}
              exact={true}
              to="/"
            >
              Home
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/users"
            >
              Users
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/gqlusers"
            >
              GQL Users
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/counter"
            >
              Counter
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/weather"
            >
              Weather
            </MenuItem>,
            <MenuMenu
              position="right"
            >
              <MenuItem
                position="right"
              >
                <LoginModal />
                <LogoutModal />
              </MenuItem>
            </MenuMenu>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Home",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Users",
              "to": "/users",
            },
            "ref": null,
            "rendered": "Users",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "GQL Users",
              "to": "/gqlusers",
            },
            "ref": null,
            "rendered": "GQL Users",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Counter",
              "to": "/counter",
            },
            "ref": null,
            "rendered": "Counter",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Weather",
              "to": "/weather",
            },
            "ref": null,
            "rendered": "Weather",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <MenuItem
                position="right"
              >
                <LoginModal />
                <LogoutModal />
              </MenuItem>,
              "position": "right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <LoginModal />,
                  <LogoutModal />,
                ],
                "position": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
